src/django/api/views/v1/production_locations.py

```diff
@@ -0,0 +1,54 @@
+from rest_framework import status
+from rest_framework.viewsets import ViewSet
+from rest_framework.response import Response
+from api.views.v1.utils import (
+    serialize_params,
+    handle_value_error,
+    handle_opensearch_exception
+)
+from api.services.search import OpenSearchService, OpenSearchServiceException
+from api.views.v1.opensearch_query_builder.opensearch_query_builder \
+    import OpenSearchQueryBuilder
+from api.views.v1.opensearch_query_builder.opensearch_query_director \
+    import OpenSearchQueryDirector
+from api.serializers.v1.production_locations_serializer \
+    import ProductionLocationsSerializer
+from api.views.v1.index_names import OpenSearchIndexNames
+
+
+class ProductionLocations(ViewSet):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.opensearch_service = OpenSearchService()
+        self.opensearch_query_builder = OpenSearchQueryBuilder()
+
+    def list(self, request):
+        if request.method == 'GET':
```

Comment:
Can we use [view decorators](https://docs.djangoproject.com/en/5.0/topics/http/decorators/#django.views.decorators.http.require_http_methods) instead of checking for the `GET` request?

Reply:
Alternatively, we can do something do a negative check here:

```python
if request.method != request.GET:
  throw MethodNotAllowed()
```

Reply:
Let's stick with `raise MethodNotAllowed()` here because forementioned decorator is generally used with function-based views, not class-based views. I've updated the code.

Reply:
@VadimKovalenkoSNF 
Do we really need this check? As I understand it, the `def list` method is considered by Django as the GET method be default.

In the case of custom methods, we need to use decorators.

Reply:
Yes, you are right. I've double checked it, and it's redundant. Code has been updated.